### ISSUE TITLE
Deprecation of one of dplyr features in package "ncappc"

### ISSUE THREAD

**Issue by MohsenSoltanifar** [tidyverse/dplyr#7014]
---

Brief description of the problem

```r
Hi All;
I am running the ncappc function in the package ncappc to estimate noncompartmental parameters in my PK clinical trial data. 
It gives an error.
Can you please guide how to fix this ?
Thanks.
---------------------------------------------------------------------
> ncappc(
+   obsFile = RawPK,
+   str1Nm = "IND",
+   str1 = c(0,1),
+   concUnit = "ng/ml",
+   timeUnit = "hours",
+   doseUnit = "ng",
+   idNmObs = "ID",
+   timeNmObs = "TAD",
+   concNmObs = "DV",
+   doseAmtNm = "DOSE",
+   onlyNCA = TRUE,
+   LambdaTimeRange = "NULL",
+   method = "linearup-logdown",
+   studyName = "PK Tutorial",
+   outFileNm = "WOW_PK_TUTORIAL",
+   extrapolate = TRUE,
+   gg_theme = theme_classic()
+ )
[1] "Note: Incorrect time range for Lambda calculation. LambdaTimeRange will not be used."
[1] "Note: EVID column is not present. EVID will not be used to process the observed data."
Error in `$<-.data.frame`(`*tmp*`, "type1", value = "Linear") : 
  replacement has 1 row, data has 0
In addition: Warning message:
The `add` argument of `group_by()` is deprecated as of dplyr 1.0.0.
ℹ Please use the `.add` argument instead.
ℹ The deprecated feature was likely used in the dplyr package.
  Please report the issue at <https://github.com/tidyverse/dplyr/issues>.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated. 

> lifecycle::last_lifecycle_warnings()
[[1]]
<warning/lifecycle_warning_deprecated>
Warning:
**The `add` argument of `group_by()` is deprecated as of dplyr 1.0.0.**
ℹ Please use the `.add` argument instead.
ℹ The deprecated feature was likely used in the dplyr package.
  Please report the issue at <https://github.com/tidyverse/dplyr/issues>.
---
Backtrace:
    ▆
 1. ├─ncappc::ncappc(...)
 2. │ └─ncappc:::estimate_nca(...)
 3. │   └─ncappc:::estimate_nca_tidy(...)
 4. │     └─pddf %>% dplyr::group_by(!!!strats, add = TRUE)
 5. ├─dplyr::group_by(., !!!strats, add = TRUE)
 6. └─dplyr:::group_by.data.frame(., !!!strats, add = TRUE)
 7.   └─dplyr::group_by_prepare(.data, ..., .add = .add, error_call = current_env())

--------------------------------------------------------------------- 
```

**Comment by philibe** [comment]
Hello,

Without reprex (reproducible example) , in particular the data `RawPK`, it could be difficult to help you.

That said the package `ncappc` seems to be 6 years old !! (read on the mirror https://github.com/cran/ncappc)

dplyr have been updated since this long time in particular `group_by()`. Now R  is in version  4.3 and the dependency of  `ncappc` about R is version 2: `Depends: R (>= 2.15.3)`  !!
https://github.com/cran/ncappc/blob/a2b7e6a1f121a6ba1a7132a5a7809b167a0815c5/DESCRIPTION#L13

Someone needs to update  `ncappc`.  Try to contact the author.
https://github.com/cran/ncappc/blob/a2b7e6a1f121a6ba1a7132a5a7809b167a0815c5/DESCRIPTION#L5

In other words: the problem does not come from dplyr but from ncappc.

PS: I'm a simple user

**Comment by DavisVaughan** [comment]
I agree with @philibe that you should try and contact the ncappc authors about this, thanks!
